### PR TITLE
Lc problem16 4 sum

### PR DESCRIPTION
**Pull request by CHAriina**
Solving the FourSum problem (day 15, item 2)

### CODE REVIEW

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        public IList<IList<int>> FourSum(int[] nums, int target)
+        {
+            List<IList<int>> result = new List<IList<int>>();
+
+            Array.Sort(nums);
+
+            for (int firstNumIndex = 0; firstNumIndex < nums.Length - 3; firstNumIndex++)
+            {
+                if (firstNumIndex > 0 && nums[firstNumIndex] == nums[firstNumIndex - 1]) continue;
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: don't volatile your style code!
'continue' should go from the new line and in curly brackets

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                for (int secondNumIndex = firstNumIndex + 1; secondNumIndex < nums.Length - 2; secondNumIndex++)
+                {
+                    if (secondNumIndex > firstNumIndex + 1 && nums[secondNumIndex] == nums[secondNumIndex - 1]) continue;
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: don't volatile your style code!
'continue' should go from the new line and in curly brackets

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+                    while (leftPointer < rightPointer)
+                    {
+                        long sum = (long)nums[firstNumIndex] + nums[secondNumIndex] + nums[leftPointer] + nums[rightPointer];
+
+                        if (sum == target)
+                        {
+                            result.Add(new List<int> { nums[firstNumIndex], nums[secondNumIndex], nums[leftPointer], nums[rightPointer] });
+
+                            while (leftPointer < rightPointer && nums[leftPointer] == nums[leftPointer + 1]) leftPointer++;
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: don't volatile your style code!
'leftPointer++' should go from the new line and in curly brackets

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                            while (leftPointer < rightPointer && nums[rightPointer] == nums[rightPointer - 1]) rightPointer--;
```
Comment (dsubotkovsky) [inline ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: don't volatile your style code!
'leftPointer--' should go from the new line and in curly brackets

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs (hunk 1 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: code has been brought to a single style

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs (hunk 3 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: code has been brought to a single style

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs (hunk 2 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: code has been brought to a single style

File: ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs (hunk 4 above)
Comment (CHAriina) [reply ScnSoft-Education/LeetCodeTraining/FourSum16/FourSum.cs]: code has been brought to a single style

**Pull request merged**